daphne/src/roles/helper.rs

```diff
@@ -216,42 +217,104 @@ pub trait DapHelper<S>: DapAggregator<S> {
         // The reason we do this is because we don't expect replays to happen but we have to guard
         // against them, as such, even though retrying is possibly very expensive, it probably
         // won't happen often enough that it matters.
-        let (out_shares_count, agg_job_resp) = {
-            let mut replayed_reports = HashSet::new();
-            let mut retry_count = 3;
-            loop {
-                if retry_count < 1 {
-                    // we need to prevent an attacker from keeping this loop running for too long,
-                    // potentialy enabling an DOS attack.
-                    return Err(DapAbort::BadRequest(
-                        "AggregationJobContinueReq contained too many replays".into(),
-                    ));
+        const RETRY_COUNT: u32 = 3;
+        let mut input_transitions = agg_job_cont_req.transitions;
+        let mut output_transitions: Vec<(ReportId, Option<TransitionVar>)> = input_transitions
```

Comment:
Let's try to think of names for these that are more evocative of their roles. Perhaps "uncommitted transitions" and "committed transitions", in the sense of "committed to the storage change"